src/main/java/server/router/RequestRouter.java

```diff
@@ -0,0 +1,36 @@
+package server.router;
+
+import java.util.Arrays;
+import java.util.List;
+import server.request.Request;
+import server.response.Response;
+import server.router.routes.HeadRequest;
+import server.router.routes.NotFound;
+import server.router.routes.Route;
+import server.router.routes.SimpleGet;
+import server.router.routes.SimpleGetWithBody;
+
+public class RequestRouter {
+  private Request request;
+  private List<Route> routes;
+
+  public Response getResponse(Request request) {
+    this.request = request;
+    routes = getRouteList();
+
+    for (Route route : routes) {
+      if (request.path().equals(route.path())) {
+        return route.processRequest();
+      }
+    }
+    return new NotFound(request).processRequest();
+  }
+
+  private List<Route> getRouteList() {
+    return Arrays.asList(
+            new SimpleGet(request),
+            new SimpleGetWithBody(request),
+            new HeadRequest(request)
```

Comment:
This looks like an OCP violation - every time we need to add a new route, we'd need to create a new class. What do you think about creating a generic class to handle these different types of routes?

Reply:
Don't know about generic classes but I'll research this!

Reply:
I don't mean `generic` as a special type of Java class, but more as one that is reusable. It looks like `SimpleGet`, `SimpleGetWithBody`, etc. all have the same methods but with different return values. Could we perhaps make a reusable class in place of having to create specific classes? A non-polished example could look like:

```java
new Path(request, "/simple_get", Arrays.asList("GET"), null)

public class Path extends Route {
  private String path;
  private List<String> methods;
  private String resource;

  public Path(Request request, String path, List<String> methods, String resource) {
    super(request);
    this.path = path;
    this.methods = methods;
    this.resource = resource;
  }

  @Override
  public String path() {
    return this.path;
  }

  // rest of the code
}
```

Reply:
Ohhhh. Nice!! Thanks for explaining that... I'll start playing around with this

Reply:
That makes so much sense the more I look at it. Thank you!!

Reply:
Looks like there was a discussion that I missed to mimic how routing is done on other frameworks, which makes sense to me and I completely overlooked that earlier. Would it be too difficult to revert these changes?

Reply:
Not a problem! Will do while I'm making these other changes

Reply:
reverted